src/emulator/app.cpp

```diff
@@ -265,28 +262,11 @@ static bool load_app_impl(Ptr<const void> &entry_point, HostState &host, const s
     find_data(host.io.title_id, host.sfo_handle, "TITLE_ID");
 
     if (host.cfg.archive_log) {
-        fs::create_directory(std::string(host.base_path) + "/logs");
-        try {
-            std::string game_title = string_utils::remove_special_chars(host.game_title);
-            const auto log_name = fmt::format("{} - [{}].log", game_title, host.io.title_id);
-#ifdef WIN32
-            wchar_t buffer[MAX_PATH];
-            GetModuleFileNameW(NULL, buffer, MAX_PATH);
-            std::string::size_type pos = std::wstring(buffer).find_last_of(L"\\\\");
-            std::wstring path = std::wstring(buffer).substr(0, pos);
-
-            if (!path.empty()) {
-                const auto full_log_path = path + L"\\" + L"\\logs\\" + string_utils::utf_to_wide(log_name);
-                logging::add_sink(full_log_path);
-            } else {
-                std::cerr << "failed to get working directory" << std::endl;
-            }
-#else
-            logging::add_sink(string_utils::utf_to_wide(log_name));
-#endif
-        } catch (const spdlog::spdlog_ex &ex) {
-            std::cerr << "File log initialization failed: " << ex.what() << std::endl;
-        }
+        const fs::path log_directory{ host.base_path + "/logs" };
+        fs::create_directory(log_directory);
+        const auto log_name{ log_directory / (host.game_title + " - [" + host.io.title_id + "].log") };
```

Comment:
you need to sanitize `host.game_title` with `string_utils::remove_special_chars` for the logging to work correctly

Reply:
Done.

I initially assumed that Boost.Filesystem removes all the special characters from the filename but it [doesn't apparently remove all of the ones mentioned in `remove_special_chars`](https://www.boost.org/doc/libs/1_70_0/libs/filesystem/doc/reference.html#Operating-system-examples) from my searching.